src/os/linux/xdg.rs

```diff
@@ -0,0 +1,412 @@
+        result_text.extend_from_slice(&text.as_bytes()[last_index_read..start]);
+
+        // We put the part we want to replace the match with
+        result_text.extend_from_slice(replace_by.as_bytes());
+
+        // Then we jump the last index to the end of the regex, ignoring the part we matched
+        last_index_read = end;
+    }
+    // At the end of the loop, put the rest of the string
+    result_text.extend_from_slice(&text.as_bytes()[last_index_read..]);
+
+    OsString::from_vec(result_text)
+}
+
+fn parse_quoted_text(
+    text: &OsStr,
+    argument: &OsStr,
+    icon: Option<&OsStr>,
+    desktop_entry_path: Option<&Path>,
+) -> OsString {
+    // We parse the escape character (\) again on the quoted text
+    let mut result = Vec::new();
+    let mut escaping = false;
+    for &c in text.as_bytes() {
+        if escaping {
+            // If escaping, then pass whatever char c is, then stop escaping
+            result.push(c);
+            escaping = false;
+        } else {
+            // If not escaping, check for whether c is escape ('\'), going into escaping
+            // mode if yes (dropping c). Otherwise just pass the c char.
+            if c == b'\\' {
+                escaping = true;
+            } else {
+                result.push(c);
+            }
+        }
+    }
+    let result = OsString::from_vec(result);
+
+    // Now we do the unquoted part
+    parse_unquoted_text(&result, argument, icon, desktop_entry_path)
+}
+
+fn parse_unquoted_text(
+    text: &OsStr,
+    argument: &OsStr,
+    icon: Option<&OsStr>,
+    desktop_entry_path: Option<&Path>,
+) -> OsString {
+    // We parse the arguments
+    // We only have one file path (not an URL). Any instance of these ones
+    // needs to be replaced by the file path in this particular case.
+    let arg_re = byte_regex!(r"%u|%U|%f|%F");
+    let result = replace_on_pattern(text, argument, arg_re);
+
+    // Then the other flags
+    let icon_replace = if let Some(icon_os_str) = icon {
+        icon_os_str
+    } else {
+        "".as_ref()
+    };
+    let result = replace_on_pattern(result, icon_replace, byte_regex!("%i"));
+
+    let desktop_entry_replace = if let Some(path) = desktop_entry_path {
+        path.as_os_str()
+    } else {
+        "".as_ref()
+    };
+    let result = replace_on_pattern(result, desktop_entry_replace, byte_regex!("%k"));
+
+    // The other % flags are deprecated so we clear them, except double percentage
+    // The spec from freedesktop does not even list what they should mean
+    let result = replace_on_pattern(result, "", byte_regex!(r"%[^%]"));
+
+    // Of course, the double percentage maps to percentage
+    let result = replace_on_pattern(&result, "%", byte_regex!("%%"));
+
+    result
+}
+
+// The exec field of the FreeDesktop entry may contain some flags that need to
+// be replaced by parameters or even other stuff. I am trying to implement it
+// all this time.
+//
+// This function kind of became a monster
```

Comment:
It's definitely monstrous, but it looks solid! Thanks a ton for implementing this.